src/js/pyodide.js

```diff
@@ -198,6 +203,35 @@ export async function loadPyodide(config) {
   Module.indexURL = baseURL;
   let packageIndexReady = initializePackageIndex(baseURL);
 
+  if(config.input) {
+    let input = [];
+    let inputIndex = -1; // -1 means that we just returned null
+    function stdin() {
+      if (inputIndex === -1) {
+        input = Module.intArrayFromString((config.input() || "") + "\n", true, 0);
+        inputIndex = 0;
+      }
+
+      if (inputIndex < input.length) {
+        let character = input[inputIndex];
+        inputIndex++;
+        return character;
+      } else {
+        inputIndex = -1;
+        return null;
+      }
+    }
```

Comment:
Is this function right? It always returns `null` once after every complete string, and it doesn't seem to allow `config.input` to respond that it doesn't have any data. I would think something more like this:
```js
    let input = [];
    let inputIndex = 0;
    let encoder = new TextEncoder();
    function stdin() {
      if(inputIndex < input.length){
        let character = input[inputIndex];
	    inputIndex++;
	    return character;
      }
      let newInput = config.input();
      if(newInput === undefined){
        return null;
      } 
      if(typeof newInput !== "string"){
      	throw new TypeError(`stdin callback returned invalid output type ${typeof newInput}, expected a string or undefined.`);
      }
      input = TextEncoder.encode(newInput + "\n");
      inputIndex = 1;
      return input[0];
    }
```

Reply:
I wrote the function with the typical behavior of `input` in mind.

So, when something from stdin is requested, the `config.input()` function gets called exactly once for every line.
To tell Emscripten that the line/user input is done, I have to return `null` after it. Otherwise a single `input()` call would result in multiple prompts until the user hits the cancel button.

Reply:
Great that makes sense. I assume it doesn't work if you return 0? If returning 0 would work, we could just say `TextEncoder.encode(newInput + "\n\x00");` in my code above?

Reply:
@hoodmane I'll admit, I haven't tried it out. It's possible that it would work, but I'd prefer to rely on the little bit of emscripten that's documented if possible

> It should return an ASCII character code when data is available, or `null` when it isn’t.
> -- https://emscripten.org/docs/api_reference/Filesystem-API.html#setting-up-standard-i-o-devices